superset/assets/src/explore/components/AdhocFilterEditPopover.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Popover, Tab, Tabs } from 'react-bootstrap';
+
+import columnType from '../propTypes/columnType';
+import adhocMetricType from '../propTypes/adhocMetricType';
+import AdhocFilter, { EXPRESSION_TYPES } from '../AdhocFilter';
+import AdhocFilterEditPopoverSimpleTabContent from './AdhocFilterEditPopoverSimpleTabContent';
+import AdhocFilterEditPopoverSqlTabContent from './AdhocFilterEditPopoverSqlTabContent';
+
+const propTypes = {
+  adhocFilter: PropTypes.instanceOf(AdhocFilter).isRequired,
+  onChange: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onResize: PropTypes.func.isRequired,
+  options: PropTypes.arrayOf(PropTypes.oneOfType([
+    columnType,
+    PropTypes.shape({ saved_metric_name: PropTypes.string.isRequired }),
+    adhocMetricType,
+  ])).isRequired,
+  datasource: PropTypes.object,
+};
+
+const startingWidth = 300;
+const startingHeight = 190;
+
+export default class AdhocFilterEditPopover extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onSave = this.onSave.bind(this);
+    this.onDragDown = this.onDragDown.bind(this);
+    this.onMouseMove = this.onMouseMove.bind(this);
+    this.onMouseUp = this.onMouseUp.bind(this);
+    this.onAdhocFilterChange = this.onAdhocFilterChange.bind(this);
+
+    this.state = {
+      adhocFilter: this.props.adhocFilter,
+      width: startingWidth,
+      height: startingHeight,
+    };
+    document.addEventListener('mouseup', this.onMouseUp);
```

Comment:
I've typically seen this in the `componentDidMount` lifecycle event but am not sure if it matters too much.